openfpga_flow/vpr_arch/k4_frac_N8_tileable_reset_softadder_register_scan_chain_nonLR_caravel_io_skywater130nm.xml

```diff
@@ -33,7 +33,15 @@
         <port name="inpad"/>
       </output_ports>
     </model>
-
+    <model name="adder_lut4">
```

Comment:
Why are we adding this model?

I already updated vpr_arch.xml as written in my earlier email. Please refer to xml with soft_adder definition in SOFA repo branch 'interface_adder': [vpr_arch xml](https://github.com/lnis-uofu/SOFA/blob/interface_adder/ARCH/vpr_arch/k4_N8_tileable_reset_softadder_register_scan_chain_nonLR_caravel_io_skywater130nm.xml)

Also review changes in openfpga xml ([openfpga xml](https://github.com/lnis-uofu/SOFA/blob/interface_adder/ARCH/openfpga_arch_template/k4_N8_reset_softadder_register_scan_chain_caravel_io_skywater130nm_fdhd_cc_openfpga.xml))

Reply:
Hi @lnsharma I am using the QLSOFA architecture for this regression test. It contains a fracturable lut4 different from the AP3 architecture. The frac_lut4 in QLSOFA has three types of outputs: lut2_out, lut3_out, lut4_out, while the frac_lut in AP3 has two types of outputs: lut2_out and lut4_out. That is why I created this adder_lut4 model

Reply:
Is there any issue in using the same name - frac_lut4 in AP3 vpr_arch xml because this is already defined as a model in this file and in case of soft_adder, we are just using it instead of introducing another model name - adder_lut4?
If you really want to introduce another model - adder_lut4, then we need to replace all instances of frac_lut4 from vpr_arch xml.

Reply:
There is no such issue in AP3 architecture, where the frac_lut4 in physical mode is the same as the frac_lut4 in the adder operating mode. The test case is just an example. I think you can keep the current yosys script for AP3 architecture. For QLSOFA, we have a different yosys script.